tools/benchmarks/llm_eval_harness/meta_eval/prepare_meta_eval.py

```diff
@@ -134,18 +174,33 @@ def change_yaml(args, base_name):
                         "WORK_DIR", str(yaml_dir)
                     )
                 )
+    # 3.2 evals dataset has a differents set of tasks from 3.1
+    # Update tasks in meta_pretrain.yaml
+    with open(args.template_dir + "/meta_pretrain.yaml", "r") as yaml_file:
+        meta_pretrain = yaml.safe_load(yaml_file)
+    if args.evals_dataset in LLAMA_3_1_PRETRAIN_EVALS:
+        meta_pretrain["task"] = ["meta_bbh", "meta_mmlu_pro_pretrain"]
+    elif args.evals_dataset in LLAMA_3_2_PRETRAIN_EVALS:
+        meta_pretrain["task"] = ["meta_mmlu"]
+    with open(args.work_dir + "/meta_pretrain.yaml", "w") as yaml_file:
+        yaml.dump(meta_pretrain, yaml_file)
+    
+    # Update tasks in meta_instruct.yaml
+    with open(args.template_dir + "/meta_instruct.yaml", "r") as yaml_file:
+        meta_instruct = yaml.safe_load(yaml_file)
+    if args.evals_dataset in LLAMA_3_1_INSTRUCT_EVALS:
+        meta_instruct["task"] = ["meta_ifeval", "meta_math_hard", "meta_gpqa_cot", "meta_mmlu_pro_instruct"]
+    elif args.evals_dataset in LLAMA_3_2_INSTRUCT_EVALS:
+        meta_instruct["task"] = ["meta_mmlu", "meta_math", "meta_gpqa"]
+    with open(args.work_dir + "/meta_instruct.yaml", "w") as yaml_file:
+        yaml.dump(meta_instruct, yaml_file)
 
 
 # copy the files and change the yaml file to use the correct model name
 def copy_and_prepare(args):
     # nltk punkt_tab package is needed
     nltk.download('punkt_tab')
-    if not os.path.exists(args.work_dir):
-        # Copy the all files, including yaml files and python files, from template folder to the work folder
-
-        copy_dir(args.template_dir, args.work_dir)
-    else:
-        print("work_dir already exists, no need to copy files")
```

Comment:
This prevents this issue:
```
(.venv) ubuntu@155-248-254-95:~/1xa100-2/llama-recipes/tools/benchmarks/llm_eval_harness/meta_eval$ python prepare_meta_eval.py --config_path ./eval_config.yaml
[nltk_data] Downloading package punkt_tab to /home/ubuntu/nltk_data...
[nltk_data]   Package punkt_tab is already up-to-date!
work_dir already exists, no need to copy files
changing ./meta_template/bbh/bbh_3shot_cot.yaml to output_path: ./work_dir/bbh/bbh_3shot_cot.yaml
changing ./meta_template/gpqa_cot/gpqa_0shot_cot.yaml to output_path: ./work_dir/gpqa_cot/gpqa_0shot_cot.yaml
changing ./meta_template/ifeval/ifeval.yaml to output_path: ./work_dir/ifeval/ifeval.yaml
changing ./meta_template/mmlu_pro/mmlu_pro_5shot_cot_instruct.yaml to output_path: ./work_dir/mmlu_pro/mmlu_pro_5shot_cot_instruct.yaml
changing ./meta_template/mmlu_pro/mmlu_pro_5shot_cot_pretrain.yaml to output_path: ./work_dir/mmlu_pro/mmlu_pro_5shot_cot_pretrain.yaml
changing ./meta_template/mmlu/mmlu.yaml to output_path: ./work_dir/mmlu/mmlu.yaml
changing ./meta_template/gpqa/gpqa_0shot.yaml to output_path: ./work_dir/gpqa/gpqa_0shot.yaml
Traceback (most recent call last):
  File "/home/ubuntu/1xa100-2/llama-recipes/tools/benchmarks/llm_eval_harness/meta_eval/prepare_meta_eval.py", line 279, in <module>
    copy_and_prepare(args)
  File "/home/ubuntu/1xa100-2/llama-recipes/tools/benchmarks/llm_eval_harness/meta_eval/prepare_meta_eval.py", line 208, in copy_and_prepare
    change_yaml(args, base_name)
  File "/home/ubuntu/1xa100-2/llama-recipes/tools/benchmarks/llm_eval_harness/meta_eval/prepare_meta_eval.py", line 170, in change_yaml
    with open(output_path, "w") as output:
FileNotFoundError: [Errno 2] No such file or directory: './work_dir/gpqa/gpqa_0shot.yaml'
```

If a user already has a work_dir, then they won't get the new tasks.

Lmk if you want a different approach though.